cedar-policy-symcc/src/symcc/decoder.rs

```diff
@@ -576,307 +584,282 @@ impl SExpr {
         }
     }
 
-    /// Decodse a literal (with only SMT constants and no bound variables).
-    pub fn decode_literal(&self, id_maps: &IdMaps) -> Result<Term, DecodeError> {
-        match self {
-            SExpr::BitVec(bv) => Ok(Term::Prim(TermPrim::Bitvec(bv.clone()))),
-            SExpr::String(s) => Ok(Term::Prim(TermPrim::String(s.clone()))),
+    /// Decodes an [`SExpr`] as an entity UID or record.
+    fn decode_entity_or_record(
```

Comment:
The changes in the rest of this file look large and I'm not sure what the motivation is.  I looked in `Decoder.lean` and it doesn't look like we have an analogue of this `decode_entity_or_record()` function on the Lean side?  What is the change doing / what's the motivation?

Reply:
In that case, could we add those additional cases to the Lean version too?  and maybe do this refactor on the Lean side as well?

Reply:
sure let me do that!

Reply:
Thanks!